### PR TITLE
Standard tests: update template

### LINKED ISSUE #2185: Standard Tests: Use by default in generated templates
## Tell us about the problem you're trying to solve
placeholder issue to use v2 tests in templates by default once the various components of test v2 are in place

### PR DESCRIPTION
**Pull request by keu**
closes #2185 
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

### CODE REVIEW

**Review by sherifnada — CHANGES_REQUESTED** [review]
`we`` should also change the README to show how to run tests, and do the same thing for the singer template

File: airbyte-integrations/connector-templates/source-python/sample_files/invalid_config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "fix-me-too": "value"
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-python/sample_files/invalid_config.json:2]: everything that a user should fill in should have a `TODO` --  in the instructions we're going to tell them to grep the project for "TODO" and apply the instructions

File: airbyte-integrations/connector-templates/source-singer/README.md.hbs [hunk 2]
```diff
@@ -99,6 +99,10 @@ docker run --rm -v $(pwd)/secrets:/secrets -v $(pwd)/sample_files:/sample_files
 1. To run additional integration tests, create a directory `integration_tests` which contain your tests and run them with `pytest integration_tests`.
    Make sure to familiarize yourself with [pytest test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery) to know how your test files and methods should be named.
 
+### Standard Tests
+1. Update standard_test_config.yml file to disable/repeat or customize behaviour/inputs of specific tests.
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-singer/README.md.hbs:103]: same comment as above

**Review by sherifnada — APPROVED** [review]
Looks good with some small comments! We should merge this with the docs changes so they are done atomically probably.

File: airbyte-integrations/connector-templates/source-python/setup.py.hbs [hunk 3]
```diff
@@ -24,12 +24,24 @@ SOFTWARE.
...
     name="source_{{snakeCase name}}",
     description="Source implementation for {{titleCase name}}.",
     author="Airbyte",
     author_email="contact@airbyte.io",
     packages=find_packages(),
-    install_requires=["airbyte-protocol", "base-python", "pytest==6.1.2"],
-    package_data={"": ["*.json"]}
+    install_requires=MAIN_REQUIREMENTS,
+    package_data={"": ["*.json"]},
+    extras_require={
```
[the remark this answers is not in the document]
Comment (keu) [reply airbyte-integrations/connector-templates/source-python/setup.py.hbs:44]: there are many envs, testing env might require quite heavy dependencies, and absolutely not needed for a connector to work. usually, it is a good thing to separate such dependencies, as well as sometimes separate dev dependencies (like tools to debug, etc)

[the remark this answers is not in the document]
Comment (sherifnada) [reply airbyte-integrations/connector-templates/source-python/setup.py.hbs:44]: Ok in this case we should also update the `README.md` testing section to say that the user should install `pip install ".[tests]"` before running tests

File: airbyte-integrations/connector-templates/source-python/build.gradle.hbs [hunk 4]
```diff
@@ -1,32 +1,14 @@
...
-    // for incremental syncing (e.g. include cursor fields, etc).
-    configuredCatalogPath = "sample_files/configured_catalog.json"
-}
-
-
 dependencies {
-    implementation files(project(':airbyte-integrations:bases:base-standard-source-test-file').airbyteDocker.outputs)
-    implementation files(project(':airbyte-integrations:bases:base-python').airbyteDocker.outputs)
+    implementation files(project(':airbyte-integrations:bases:standard-test').airbyteDocker.outputs)
+    implementation files(project(':airbyte-integrations:bases:base-singer').airbyteDocker.outputs)
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-python/build.gradle.hbs:13]: why is a python source depending on singer?

File: airbyte-integrations/connector-templates/source-singer/README.md.hbs [hunk 5]
```diff
@@ -93,19 +73,47 @@ docker run --rm -v $(pwd)/secrets:/secrets airbyte/source-{{dashCase name}}-sing
...
-1. To run additional integration tests, create a directory `integration_tests` which contain your tests and run them with `pytest integration_tests`.
+## Testing
    Make sure to familiarize yourself with [pytest test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery) to know how your test files and methods should be named.
+First install test dependecies into your virtual environment:
+```
+pip install .[test]
+```
+### Unit Tests
+To run unit tests locally, from the connector directory run:
+```
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-singer/README.md.hbs:84]: same comments from other readme

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 6]
```diff
@@ -26,7 +26,11 @@ See all the test cases and their description in [Standard Source Tests](standard
 
 ### Setting up standard tests for your connector
 
-Standard tests are typically run from a docker container. The default standard test runner is the File-based Standard Test suite, which gets its name because its inputs are passed as files via Docker volume mounts. This is the simplest way to run the standard test suite: the only requirements are that you place its input files inside your connector's directory, and to pass the paths to those input files as arguments to the Gradle plugin required to invoke it. This is setup by default inside the `build.gradle` file on all connectors generated from templates.
+Standard tests are typically run from a docker container.
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:29]: I think we should remove all references to the v1 test suite into a separate md page (lets say maybe `Standard Tests (Legacy)`) and link to it in this page saying something like `If you are looking for reference documentation for the deprecated first version of test suites, see <link>`. 

The current md file has a little bit of split-brain between v1 and v2. I would rather we start fresh for a v2 document and feel free to copy anything from the v1 document that you like.

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 7]
```diff
@@ -50,6 +54,24 @@ These inputs are all described in the [Airbyte Specification](../../architecture
...
 
+## Airbyte's Standard Tests (v2) - Source Acceptance Tests
+
+Unfortunately Standard Tests (v1) has some problems: 
+* Its tests are opaque to a contributor, difficult to debug when something goes wrong
+* It tries to encompass testing all features/edge cases in one test case which has made it difficult to configure and its behavior hard to understand
+
+Source Acceptance Tests tries to address these issues. These are the differences from v1:
+* configured with YAML file, forget about Gradle
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:64]:
```suggestion
* configured with YAML file and run via Docker or pytest
```

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 8]
```diff
@@ -50,6 +54,24 @@ These inputs are all described in the [Airbyte Specification](../../architecture
...
+* individual tests can be disabled in the config file
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:65]:
```suggestion
* test suites can be enabled/disabled via config 
```

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 9]
```diff
@@ -50,6 +54,24 @@ These inputs are all described in the [Airbyte Specification](../../architecture
...
+* individual tests can have different inputs and run multiple time with different sets of inputs
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:66]:
```suggestion
* test suites can run multiple time with different sets of inputs
```

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md (hunk 8 above)
Comment (sherifnada) [reply docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:65]: I think `individual tests` makes it sound like test cases not test suites (e.g: incremental is a suite)

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 10]
```diff
@@ -50,6 +54,24 @@ These inputs are all described in the [Airbyte Specification](../../architecture
...
+* each test has a timeout
+* each test has stores all inputs and outputs
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:68]:
```suggestion
* each test stores all inputs and outputs, which can be used for debugging failures
```

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 11]
```diff
@@ -50,6 +54,24 @@ These inputs are all described in the [Airbyte Specification](../../architecture
...
+* Its tests are opaque to a contributor, difficult to debug when something goes wrong
+* It tries to encompass testing all features/edge cases in one test case which has made it difficult to configure and its behavior hard to understand
+
+Source Acceptance Tests tries to address these issues. These are the differences from v1:
+* configured with YAML file, forget about Gradle
+* individual tests can be disabled in the config file
+* individual tests can have different inputs and run multiple time with different sets of inputs
+* each test has a timeout
+* each test has stores all inputs and outputs 
+* use pytest as a test runner
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:69]:
```suggestion
* uses pytest as a test runner
```

File: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md [hunk 12]
```diff
@@ -50,6 +54,24 @@ These inputs are all described in the [Airbyte Specification](../../architecture
...
+* can be run manually, with interactive debugger, custom plugins, last failed tests, etc - thanks to pytest
+* the output is easy to read, good bye Java stacktrace - thanks to pytest
```
Comment (sherifnada) [inline docs/contributing-to-airbyte/building-new-connector/testing-connectors.md:71]: Feels like this is "apologizing" for mistakes in the past but most people will not have used the previous one so we should focus on their experience

```suggestion
```

**Review by sherifnada — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connector-templates/source-python/acceptance-test.sh [hunk 13]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env sh
+CWD=$(pwd)
```
[the remark this answers is not in the document]
Comment (keu) [reply airbyte-integrations/connector-templates/source-python/acceptance-test.sh:2]: I have tested it in OS X and Ubuntu, in both cases the current directory is still original

File: airbyte-integrations/connector-templates/source-python/build.gradle.hbs (hunk 4 above)
Comment (keu) [reply airbyte-integrations/connector-templates/source-python/build.gradle.hbs:13]: probably copy-paste mistake

File: airbyte-integrations/connector-templates/source-python/build.gradle.hbs [hunk 14]
```diff
@@ -1,32 +1,14 @@
 plugins {
     id 'airbyte-python'
     id 'airbyte-docker'
-    id 'airbyte-standard-source-test-file'
+    id 'airbyte-source-acceptance-test'
 }
 
 airbytePython {
-    moduleDirectory 'source_{{snakeCase name}}'
+    moduleDirectory 'source_{{snakeCase name}}_singer'
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-python/build.gradle.hbs:8]:
```suggestion
    moduleDirectory 'source_{{snakeCase name}}'
```

File: airbyte-integrations/connector-templates/source-python/README.md.hbs [hunk 15]
```diff
@@ -80,19 +73,54 @@ docker run --rm -v $(pwd)/secrets:/secrets airbyte/source-{{dashCase name}}:dev
...
+```
 
 ### Integration Tests
-1. From the airbyte project root, run `./gradlew :airbyte-integrations:connectors:source-{{dashCase name}}:integrationTest` to run the standard integration test suite.
-1. To run additional integration tests, place your integration tests in a new directory `integration_tests` and run them with `python -m pytest -s integration_tests`.
-   Make sure to familiarize yourself with [pytest test discovery](https://docs.pytest.org/en/latest/goodpractices.html#test-discovery) to know how your test files and methods should be named.
+There are two types of integration tests: Acceptance Tests (Airbyte's test suite for all source connectors) and custom integration tests (which are specific to this connector).
+#### Acceptance Tests
+1. Customize `acceptance-test-config.yml` file to configure tests. See [Source Acceptance Tests](source-acceptance-tests.md) for more information.
+2. From the connector root, run `./acceptance-test-python.sh`
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-python/README.md.hbs:92]:
```suggestion
2. From the connector root, run `./acceptance-test-docker.sh` or `./acceptance-test-python.sh` 
```

Comment (sherifnada) [reply airbyte-integrations/connector-templates/source-python/README.md.hbs:92]: FYI added this because of this issue: https://github.com/airbytehq/airbyte/issues/3102

File: airbyte-integrations/connector-templates/source-python/README.md.hbs [hunk 16]
```diff
@@ -80,19 +73,54 @@ docker run --rm -v $(pwd)/secrets:/secrets airbyte/source-{{dashCase name}}:dev
...
+./gradlew :airbyte-integrations:connectors:source-{{dashCase name}}:SourceAcceptanceTest
+```
+To run custom integration tests run:
+```
+./gradlew :airbyte-integrations:connectors:source-{{dashCase name}}:IntegrationTest
+```
 
 ## Dependency Management
 All of your dependencies should go in `setup.py`, NOT `requirements.txt`. The requirements file is only used to connect internal Airbyte dependencies in the monorepo for local development.
+We split dependencies between two groups, dependencies that are:
```
Comment (sherifnada) [inline airbyte-integrations/connector-templates/source-python/README.md.hbs:116]:
```suggestion
We split dependencies between two groups:
```

**Review by sherifnada — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]
